File: CS/WinWebSolution.Module/Contacts.cs
using System;
using DevExpress.Xpo;
using System.ComponentModel;
using DevExpress.Data.Filtering;
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;

namespace WinWebSolution.Module {
    [DefaultClassOptions]
    [System.ComponentModel.DisplayName("Contacts")]
    public abstract class ContactBase : Person {
        public ContactBase(Session session) : base(session) { }
        private bool activeCore;
        public bool Active {
            get { return activeCore; }
            set { SetPropertyValue("Active", ref activeCore, value); }
        }
    }
    public class ContactType1 : ContactBase {
        public ContactType1(Session session) : base(session) { }
        private string contactType1PropertyCore;
        public string ContactType1Property {
            get { return contactType1PropertyCore; }
            set { SetPropertyValue("ContactType1Property", ref contactType1PropertyCore, value); }
        }
    }
    public class ContactType2 : ContactBase {
        public ContactType2(Session session) : base(session) { }
        private string contactType2PropertyCore;
        public string ContactType2Property {
            get { return contactType2PropertyCore; }
            set { SetPropertyValue("ContactType2Property", ref contactType2PropertyCore, value); }
        }
    }
}

File: VB/WinWebSolution.Module/Contacts.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports System.ComponentModel
Imports DevExpress.Data.Filtering
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl

Namespace WinWebSolution.Module
	<DefaultClassOptions, System.ComponentModel.DisplayName("Contacts")> _
	Public MustInherit Class ContactBase
		Inherits Person
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private activeCore As Boolean
		Public Property Active() As Boolean
			Get
				Return activeCore
			End Get
			Set(ByVal value As Boolean)
				SetPropertyValue("Active", activeCore, value)
			End Set
		End Property
	End Class
	Public Class ContactType1
		Inherits ContactBase
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private contactType1PropertyCore As String
		Public Property ContactType1Property() As String
			Get
				Return contactType1PropertyCore
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ContactType1Property", contactType1PropertyCore, value)
			End Set
		End Property
	End Class
	Public Class ContactType2
		Inherits ContactBase
		Public Sub New(ByVal session As Session)
			MyBase.New(session)
		End Sub
		Private contactType2PropertyCore As String
		Public Property ContactType2Property() As String
			Get
				Return contactType2PropertyCore
			End Get
			Set(ByVal value As String)
				SetPropertyValue("ContactType2Property", contactType2PropertyCore, value)
			End Set
		End Property
	End Class
End Namespace